airflow-core/src/airflow/models/variable.py

```diff
@@ -225,18 +219,20 @@ def set(
             )
             return
 
+        # check if the secret exists in the custom secrets' backend.
+        Variable.check_for_write_conflict(key=key)
         if serialize_json:
             stored_value = json.dumps(value, indent=2)
         else:
             stored_value = str(value)
-
-        Variable.delete(key, session=session)
-        session.add(Variable(key=key, val=stored_value, description=description))
-        session.flush()
-        # invalidate key in cache for faster propagation
-        # we cannot save the value set because it's possible that it's shadowed by a custom backend
-        # (see call to check_for_write_conflict above)
-        SecretCache.invalidate_variable(key)
+        with create_session() as session:
```

Comment:
```suggestion
        if session is not None:
            ctx = contextlib.nullcontext(session)
        else:
            ctx = create_session()
        with ctx as session:
```

That way we can still use the passed in session if there is one?

Reply:
Yep can do this

Reply:
Handled it this way